### ISSUE TITLE
Nextjs compatibility

### ISSUE THREAD

**Issue by RamiroPastor** [ionos-deploy-now/ionos-deploy-now#32]
# Feature request

Implement compatibility with Next.js

## Description

When using the Deploy Now user interface, I can choose which build commands to run, but then my only option is to choose an `output path`, but apparently this won't work for Next.js (https://nextjs.org/docs/going-to-production) because the server must be started with `npm run start` and it does not work as a build command.

## Value

I estimate the value of this feature over 9000% because I just cannot deploy my Nextjs app under the current conditions. So the only possible way to solve this problem is cancelling my Static Project subscription and going to vercel.com to host this app.

## Approach

How would you implement this feature if you were us?

I would just destroy every single file related to the user interface of the Deploy Now site, and start from zero. :)

**Comment by Robert95Sch** [comment]
Hi @RamiroPastor,

thanks for your feedback! I appreciate that even though you seem deeply unsatisfied with the user interface, you still provide feedback about what we can improve. 

Currently, Deploy Now deploys to Apache webservers that unfortunately do not support Node.js at runtime. This is why we only support [Static HTML Exports with Next.js](https://nextjs.org/docs/advanced-features/static-html-export). 

We've already received some [requests](https://github.com/ionos-deploy-now/ionos-deploy-now/issues/8) to support Node.js. We see a lot of value in supporting Node.js environments, however, this is no easy step for us. We will do our best and keep you updated. As soon as we support Node.js, we of course would improve the setup process for Next.js as well.

Thanks a lot,
Robert

**Comment by RamiroPastor** [comment]
Well, if I understand correctly the first paragraph of the "Static HTML Exports with Next.js":

```
next export allows you to export your Next.js application to static HTML, which can be run standalone without the need of a Node.js server. It is recommended to only use next export if you don't need any of the unsupported features requiring a server
```

It seems that this deploy method does not fulfill my requirements, because the only reason I have for using Nextjs instead of plain React is because *I need a server*, to send emails for example.

**Comment by Robert95Sch** [comment]
Yes you are right! You need a server that is able to run JavaScript, which we currently can't provide. The only server-side scripting language that is currently supported in Deploy Now is PHP. 

We wish you all the best with your project :)
